lib/gears/geometry.lua

```diff
@@ -76,4 +76,76 @@ function gears.geometry.rectangle.get_by_coord(list, x, y)
     end
 end
 
+--- Return true whether rectangle B is in the right direction
+-- compared to rectangle A.
+-- @param dir The direction.
+-- @param gA The geometric specification for rectangle A.
+-- @param gB The geometric specification for rectangle B.
+-- @return True if B is in the direction of A.
+local function is_in_direction(dir, gA, gB)
+    if dir == "up" then
+        return gA.y > gB.y
+    elseif dir == "down" then
+        return gA.y < gB.y
+    elseif dir == "left" then
+        return gA.x > gB.x
+    elseif dir == "right" then
+        return gA.x < gB.x
+    end
+    return false
+end
+
+--- Calculate distance between two points.
+-- i.e: if we want to move to the right, we will take the right border
+-- of the currently focused screen and the left side of the checked screen.
+-- @param dir The direction.
+-- @param _gA The first rectangle.
+-- @param _gB The second rectangle.
+-- @return The distance between the screens.
+local function calculate_distance(dir, _gA, _gB)
+    local gAx = _gA.x
+    local gAy = _gA.y
+    local gBx = _gB.x
+    local gBy = _gB.y
+
+    if dir == "up" then
+        gBy = _gB.y + _gB.height
+    elseif dir == "down" then
+        gAy = _gA.y + _gA.height
+    elseif dir == "left" then
+        gBx = _gB.x + _gB.width
+    elseif dir == "right" then
+        gAx = _gA.x + _gA.width
```

Comment:
For future-me: I think this needs a `-1` at the end (however, this won't ever make a difference, I think)